### PR TITLE
fix(replication) gcr deletion and tag deletion

### LINKED ISSUE #12568: The deletion replication of GCR is not work in harbor 2.x
When replicate a deletion operation to GCR, it will fail with error
```
failed to delete the manifest of artifact eminent-nation-87317/replication-ci/mybusybox:test on the destination registry: http status code: 400, body: {"errors":[{"code":"GOOGLE_MANIFEST_DANGLING_TAG","message":"Manifest is still referenced by tag: test"}]}
```

steps:
1. create a event-based replication rule with delete remote feature to Google GCR
2. delete the artifact, it will replicate fail

### PR DESCRIPTION
**Pull request by bitsf**
fix #12568 #12818

### CODE REVIEW

File: src/pkg/registry/client.go [hunk 1]
```diff
@@ -93,6 +93,8 @@ type Client interface {
 	// is used to specify whether the destination artifact will be overridden if
 	// its name is same with source but digest isn't
 	Copy(srcRepository, srcReference, dstRepository, dstReference string, override bool) (err error)
+	// Do the request
```
Comment (ywk253100) [inline src/pkg/registry/client.go:96]: Could you add more comment to explain why this method is needed?

File: src/pkg/registry/client.go [hunk 2]
```diff
@@ -93,6 +93,8 @@ type Client interface {
...
+	Do(req *http.Request) (*http.Response, error)
```
Comment (steven-zou) [inline src/pkg/registry/client.go:97]: It seems a common HTTP request handle function, is this the right place to add this func? 
It's very different with other functions in this interface.

**Review by steven-zou — COMMENTED** [review]
commented

File: src/pkg/registry/client.go (hunk 2 above)
Comment (bitsf) [reply src/pkg/registry/client.go:97]: yes, I have to add this in order to call it in adapter

**Review by ywk253100 — APPROVED** [review]

File: src/pkg/registry/client.go [hunk 3]
```diff
@@ -93,6 +93,8 @@ type Client interface {
 	// is used to specify whether the destination artifact will be overridden if
 	// its name is same with source but digest isn't
 	Copy(srcRepository, srcReference, dstRepository, dstReference string, override bool) (err error)
+	// Do this is just call the underlying method do(), usually this method should not be called directly.
```
Comment (steven-zou) [inline src/pkg/registry/client.go:96]:
```suggestion
	// Do send generic HTTP requests to the target registry service
```

File: src/pkg/registry/client.go (hunk 2 above)
Comment (steven-zou) [reply src/pkg/registry/client.go:97]: @bitsf @ywk253100 

Let's raise a debt issue to track the refactor work for the future.

**Review by steven-zou — CHANGES_REQUESTED** [review]
do changes

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**